src/components/NavBar/LoginSignIn/LogInModal/LoginModal.jsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import ReactDom from 'react-dom';
+
+import { AiOutlineClose } from 'react-icons/ai';
+import Login from '../LogInComponent';
+import SignIn from '../SignInComponent';
+
+import * as Styled from './LoginModal.styled';
+
+const LoginModal = ({ closeModal }) => {
+  const [isNewUser, setIsNewUser] = useState(false);
+  const [buttonMessage, setButtonMessage] = useState('');
+
+  const ModalContent = isNewUser ? SignIn : Login;
```

Comment:
With this ternary you know which content will be displayed for your modal, but I find it odd that you needed to pass the `setButtonMessage` to it to set a value on this component. Maybe that message should be part of your modal as well, and pass the message by prop, then render the message inside the `ModalContent`.